### PR TITLE
Added possibility to select cycle and category.

### PR DESCRIPTION
**Pull request by ewelina-slepko**

### CODE REVIEW

File: src/app/home/layouts/budgets-form/budgets-form.component.html [hunk 1]
```diff
@@ -7,56 +7,64 @@ <h2 class="header">Budgets</h2>
...
-        <i class="fas fa-circle circle"></i>
-        <i class="fas fa-plus add-category-icon"></i>
+
+      <h4 class="category-header">Category</h4>
+      <div class="categories-container">
+
+        <div *ngFor="let category of categories">
+          <i [class]="category.isSelected ?
+          'fas fa-circle category-circle selected-category-color' :
+          'fas fa-circle category-circle standard-category-color'"
```
Comment (Rementh) [inline src/app/home/layouts/budgets-form/budgets-form.component.html:50]: Do not repeat common classes  for each condition. In the ternary expression, bind only classes that differ, the rest just put as a separate un-bound class property.

File: src/app/home/layouts/budgets-form/budgets-form.component.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import {Component, OnInit} from '@angular/core';
 import {InitialSettingsService} from '../../cards/initial-settings/initial-settings.service';
 import {basicAnimation} from '../../../shared/animation';
+import {NgForm} from '@angular/forms';
+import {Categories, CategoryDto, CyclesDict, CyclesDto} from './dtos';
```
Comment (Rementh) [inline src/app/home/layouts/budgets-form/budgets-form.component.ts:5]: Why are those capitalized? I assume those are just data objects, and in that case, it should follow the camel case convention. Capitalized names are mostly used for classes, interfaces and types.

File: src/app/home/layouts/budgets-form/budgets-form.component.ts [hunk 3]
```diff
@@ -24,11 +31,26 @@ export class BudgetsFormComponent implements OnInit {
     this.initialSettingsService.sendCurrentStepInfo(3);
   }
 
+  selectCycle(cycle: CyclesDto) {
+    this.cycles.map(element => element.isSelected = false);
```
Comment (Rementh) [inline src/app/home/layouts/budgets-form/budgets-form.component.ts:35]: Do not use map if you don't need the mapped output. For simple array mutation, just go with foreach.

File: src/app/home/layouts/budgets-form/budgets-form.component.ts [hunk 4]
```diff
@@ -24,11 +31,26 @@ export class BudgetsFormComponent implements OnInit {
...
+    this.selectedCycle = cycle.name;
+    cycle.isSelected = true;
+  }
+
+  selectCategory(category: CategoryDto) {
+    this.categories.map(element => element.isSelected = false);
```
Comment (Rementh) [inline src/app/home/layouts/budgets-form/budgets-form.component.ts:41]: Do not use map if you don't need the mapped output. For simple array mutation, just go with foreach.

**Comment by ewelina-slepko** [comment]
Done!

**Pull request merged**